server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/ApplicationServiceImpl.java

```diff
@@ -283,15 +283,15 @@ private NotificationSchemaDto createDefaultNotificationSchema(String appId, Stri
     private LogSchemaDto createDefaultLogSchema(String appId, String createdUsername) {
         LogSchemaDto schema = new LogSchemaDto();
         schema.setApplicationId(appId);
-        DataSchema defSchema =  new KaaSchemaFactoryImpl().createDataSchema(getStringFromFile(DEFAULT_LOG_SCHEMA_FILE, ApplicationServiceImpl.class));
-        if (!defSchema.isEmpty()) {
-            schema.setSchema(defSchema.getRawSchema());
-        } else {
-            throw new RuntimeException("Can't read default log schema."); //NOSONAR
-        }
+        CTLSchemaDto ctlSchema = ctlService.getOrCreateEmptySystemSchema(createdUsername);
+        schema.setCtlSchemaId(ctlSchema.getId());
         schema.setName(DEFAULT_SCHEMA_NAME);
         schema.setCreatedUsername(createdUsername);
-        return logSchemaService.saveLogSchema(schema);
+        schema = logSchemaService.saveLogSchema(schema);
+        if (schema == null) {
+            throw new RuntimeException("Can't save default log schema "); //NOSONAR
```

Comment:
This is wrong anyway, if you can not save default schema - throw this exception inside logSchemaService.saveLogSchema